libqtile/scripts/launch.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2025, elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the \"Software\"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Command-line wrapper to launch process from within qtile
+"""
+
+import argparse
+import os
+
+
+def launch(opts) -> None:
+    fd = opts.fd
+
+    if fd > 0:
+        # Read the socket but we don't care what the contents is as it
+        # will be empty until qtile writes to it
+        os.read(fd, 4)
```

Comment:
It feels like we could check for your sentinel string here. I'm wondering if the source of flakiness is that this is somehow a nonblocking socket by default, so this is just giving 0?

Reply:
From my tests, `os.read` blocks when reading from a socket created by `socket.socketpair()`.

Reply:
hm, ok. unclear why it's flaky then...

Reply:
Let me add some debugging and see if we can get some clues.

Reply:
Ha. So everything passed...

Reply:
Yup. Looks like window is launching before rule has been added.

Reply:
Sometimes I find inserting manual sleep()s to trigger things in a certain order works for debugging. I can't look now, but hopefully I can look later this week.

Reply:
Latest failure showed that the `client_new` hook wasn't called to run the rules on the new window.

Reply:
@tych0 I think what's happening is that qtile is sending the sentinel but `qtile launch` isn't ready to read it yet. 

Am I right in thinking that we need to be listening for data before it's sent? The data doesn't persist until it's been read?